presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt

```diff
@@ -129,12 +132,9 @@ fun EventScreen(
 
                             eventScreenEvent(EventScreenEvent.OnShowAlarmReminderDropdown(shouldOpen = false))
 
-                            /** TODO Using mock data to set the Alarm Monitor, here use the eventScreenState start time date */
-                            val alarmItem = AlarmItem(
-                                ZonedDateTime.now().plusSeconds(30L),
-                                "THIS IS THE MESSAGE FROM THE ALARM REMINDER"
-                            )
-                            eventScreenEvent(EventScreenEvent.OnScheduleAlarmReminder(alarmItem))
+                            setupAlarmReminderItem(eventScreenState = eventScreenState, item = item, context)?.let { alarmItem ->
+                                eventScreenEvent(EventScreenEvent.OnScheduleAlarmReminder(alarmItem))
+                            }
```

Comment:
@philipplackner @MKiperszmid If something goes wrong then the `setupAlarmReminderItem` will return a nullable AlarmItem. Is this ok?

Reply:
This does not seem to be the place where you want to schedule alarms. As far as I understand, this schedules a new alarm every time the user selects a different reminder option, but what if they then decide to not save the item? You should be able to do all the alarm logic in the ViewModel itself, the UI doesn't need to trigger it.